module_evaluations/TCR-25_2023-04-19.MD

```diff
@@ -5,65 +5,66 @@ When performing a technical evaluation of a module, create a copy of this docume
-* [ ] The module responds with a tenant's content based on x-okapi-tenant header
-* [ ] Standard GET `/admin/health` endpoint returning a 200 response
-  * -_note: read more at https://wiki.folio.org/display/DD/Back+End+Module+Health+Check+Protocol_
-* [ ] High Availability (HA) compliant
-  * Possible red flags:
-    * Connection affinity / sticky sessions / etc. are used
-    * Local container storage is used
-    * Services are stateful
-* [ ] Module only uses infrastructure / platform technologies on the [officially approved technologies](https://wiki.folio.org/display/TC/Officially+Supported+Technologies) list.
-  * _e.g. PostgreSQL, ElasticSearch, etc._
+* [x] ~~Module's repository includes a compliant Module Descriptor~~
+* [x] ~~Module includes executable implementations of all endpoints in the provides section of the Module Descriptor~~
+* [x] ~~Environment vars are documented in the ModuleDescriptor~~
+* [x] ~~If a module provides interfaces intended to be consumed by other FOLIO Modules, they must be defined in the Module Descriptor "provides" section~~
+* [x] ~~All API endpoints are documented in RAML or OpenAPI~~
+* [x] ~~All API endpoints protected with appropriate permissions as per the following guidelines and recommendations, e.g. avoid using `*.all` permissions, all necessary module permissions are assigned, etc.~~
+* [x] ~~Module provides reference data (if applicable), e.g. if there is a controlled vocabulary where the module requires at least one value~~
+* [x] ~~If provided, integration (API) tests must be written in an [officially approved technology](https://wiki.folio.org/display/TC/Officially+Supported+Technologies)~~
+* [x] ~~Data is segregated by tenant at the storage layer~~
+* [x] ~~The module doesn't access data in DB schemas other than its own and public~~
+* [x] ~~The module responds with a tenant's content based on x-okapi-tenant header~~
+* [x] ~~Standard GET `/admin/health` endpoint returning a 200 response~~
+* [x] ~~High Availability (HA) compliant~~
+* [x] ~~Module only uses infrastructure / platform technologies on the [officially approved technologies](https://wiki.folio.org/display/TC/Officially+Supported+Technologies) list.~~
 
 ## TCR Process Improvements
-[_Please include here any suggestions that you feel might improve the TCR Processes._]
+* I'm not confident the "Shared/Common" criterion "Installation documentation is included" makes sense for UI modules. Maybe it just needs an "if relevant" suffix? The closest thing I can think of is configuration instructions for things like ui-eholdings, but that really isn't "installation" documentation.
+* I am not especially comfortable assessing Apache 2 license compatibility. It would be helpful to have a list of common known-good licenses. Additionally, there are tools that can assist with finding licenses in dependencies. It may be worth mentioning or providing instructions for them:

```

Comment:
@vijay-eis raised the same point about the difficulty of assessing Apache 2 license compatibility on a recent TCR review.  Vijay, I think there was a resource you used?

Reply:
[This](https://github.com/folio-org/tech-council/blob/master/module_evaluations/TCR-21_2022-12-22.MD#evaluation-notes) is what Vijay suggested:
```
mvn  license:add-third-party
cat target/generated-sources/license/THIRD-PARTY.txt  | grep -v Apache | grep -v MIT | grep -v Eclipse | grep -v BSD
```

This will likely gives a degree of filtering out of (likely) known compatibility. Any other licences would likely require manual analysis.

This [document|https://www.apache.org/legal/resolved.html#category-a] might be a starting point for which licences are compatible (this does not include all BSD, one of which is actively considered incompatible)